src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceFunctions.kt

```diff
@@ -20,22 +20,45 @@ data class SentenceChanges(
   val isMaterial: Boolean,
 )
 
-fun Licence.getSentenceChanges(newSentence: UpdateSentenceDatesRequest, newLsd: LocalDate?): SentenceChanges {
-  val lsdChanged = nullableDatesDiffer(newLsd, this.licenceStartDate)
-  val ledChanged = nullableDatesDiffer(newSentence.licenceExpiryDate, this.licenceExpiryDate)
-  val sedChanged = nullableDatesDiffer(newSentence.sentenceEndDate, this.sentenceEndDate)
-  val tussdChanged =
-    nullableDatesDiffer(newSentence.topupSupervisionStartDate, this.topupSupervisionStartDate)
-  val tusedChanged =
-    nullableDatesDiffer(newSentence.topupSupervisionExpiryDate, this.topupSupervisionExpiryDate)
-  val prrdChanged = nullableDatesDiffer(newSentence.postRecallReleaseDate, this.postRecallReleaseDate)
-  val hdcadChanged = this is HdcLicence &&
-    nullableDatesDiffer(
-      newSentence.homeDetentionCurfewActualDate,
-      this.homeDetentionCurfewActualDate,
-    )
+fun SentenceDetail.toSentenceDates() = SentenceDates(
+  conditionalReleaseDate = conditionalReleaseOverrideDate ?: conditionalReleaseDate,
+  actualReleaseDate = confirmedReleaseDate,
+  sentenceStartDate = sentenceStartDate,
+  sentenceEndDate = sentenceExpiryOverrideDate ?: sentenceExpiryDate,
+  licenceExpiryDate = licenceExpiryOverrideDate ?: licenceExpiryDate,
+  topupSupervisionStartDate = topupSupervisionStartDate,
+  topupSupervisionExpiryDate = topupSupervisionExpiryOverrideDate ?: topupSupervisionExpiryDate,
+  postRecallReleaseDate = postRecallReleaseOverrideDate ?: postRecallReleaseDate,
+  homeDetentionCurfewEligibilityDate = homeDetentionCurfewEligibilityDate,
+  homeDetentionCurfewActualDate = homeDetentionCurfewActualDate,
+  homeDetentionCurfewEndDate = homeDetentionCurfewEndDate,
+)
```

Comment:
This logic is lifted from [here](https://github.com/ministryofjustice/create-and-vary-a-licence/blob/a43bc81223260f8068517087c4bca37f08edbd3b/server/listeners/eventHandlers/prisonEvents/datesChangedEventHandler.ts#L81-L105)

Reply:
This is delightfully neater